packages/core/src/libraries/hook/index.ts

```diff
@@ -188,6 +190,48 @@ export const createHookLibrary = (queries: Queries) => {
     await sendWebhooks(webhooks, consoleLog);
   };
 
+  /**
+   * Trigger data hooks with the given data mutation context. All context objects will be used to trigger hooks.
+   */
+  const triggerExceptionHooks = async (
+    consoleLog: ConsoleLog,
+    contextManager: ExceptionHookContextManager
+  ) => {
+    if (contextManager.contextArray.length === 0) {
+      return;
+    }
+
+    const found = await findAllHooks();
```

Comment:
nit: The findAllHooks method might be executed multiple times within a single request. Maybe we should add a per-request-level cache?

Reply:
[A new Linear task](https://linear.app/silverhand/issue/LOG-12471/add-a-request-level-cache-mechanism-to-webhook-middleware) is created to track this change request.